erpnext/accounts/doctype/subscription/subscription.py

```diff
@@ -542,15 +542,17 @@ def process_for_past_due_date(self):
 			else:
 				self.set_status_grace_period()
 
-			if getdate() > getdate(self.current_invoice_end):
-				self.update_subscription_period(add_days(self.current_invoice_end, 1))
+			update_subscription = True if getdate() > getdate(self.current_invoice_end) else False
```

Comment:
Any reason why this condition cannot be shifted below? Is `current_invoice_end` changing in the meantime?

Reply:
Yes, exactly. That was causing the issue.

Reply:
The value of self.current_invoice_end should only update once the invoice is created not in between. This causes a mismatch in the condition and results in skipping an invoice creation.